### PR TITLE
[ACS-5877] Migrate widget-visibility service do date-fns

### PR DESCRIPTION
**Pull request by arohilaGL**
**Please check if the PR fulfills these requirements**

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [ ] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [x] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)
The service was earlier using moment.js

**What is the new behaviour?**

- migrate widget visibility service to date-fns
- improved typings for code

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

**Review by rbahirsheth — APPROVED** [review]

**Review by eromano — CHANGES_REQUESTED** [review]
this modify is a Braking cahnge

File: lib/core/src/lib/form/services/widget-visibility.service.ts [hunk 1]
```diff
@@ -134,7 +134,7 @@ export class WidgetVisibilityService {
         } else if (visibilityObj.rightType === WidgetTypeEnum.field) {
             valueFound = this.getFormValue(form, visibilityObj.rightValue);
         } else {
-            if (moment(visibilityObj.rightValue, 'YYYY-MM-DD', true).isValid()) {
+            if (isMatch(visibilityObj.rightValue, 'yyyy-MM-dd')) {
```
Comment (DenysVuika) [inline lib/core/src/lib/form/services/widget-visibility.service.ts:137]: this value comes from the backend server and should remain the same not to break existing forms

**Review by DenysVuika — CHANGES_REQUESTED** [review]

File: lib/core/src/lib/form/services/widget-visibility.service.ts (hunk 1 above)
Comment (arohilaGL) [reply lib/core/src/lib/form/services/widget-visibility.service.ts:137]: It is a condition check. No values are being changed. What is it that  you suggest I should do with it?

**Comment by arohilaGL** [comment]
The PR requires BE changes first.

File: lib/core/src/lib/form/services/widget-visibility.service.ts (hunk 1 above)
Comment (DenysVuika) [reply lib/core/src/lib/form/services/widget-visibility.service.ts:137]: the right value needs to be converted from moment probably

**Review by DenysVuika — CHANGES_REQUESTED** [review]

**Pull request merged**